### PR TITLE
RDK-29571 - Add Dobby specs to LEGACY_COMPONENTS

### PR DESCRIPTION
**Pull request by Anttikyll**
Dobby spec functionality (and consequently DobbyBundleGenerator) are
no longer built without LEGACY_COMPONENTS=ON.

With LEGACY_COMPONENTS=OFF, ctemplate is no longer needed.

### CODE REVIEW

**Comment by Anttikyll** [comment]
Works fine on Sky XI1 build with LEGACY_COMPONENTS=ON.

Building an image with LEGACY_COMPONENTS=OFF to check that launching from bundle still works.

**Comment by Anttikyll** [comment]
`exec()` doesn't work with LEGACY_COMPONENTS=OFF, looking into it.

**Comment by Anttikyll** [comment]
Fixed exec() issue caused by container states not being updated correctly. Basic container usage via bundles seems to work fine on the Xi1 with LEGACY_COMPONENTS=OFF.

EPG obviously doesn't work since specs are disabled.

File: .gitignore [hunk 1]
```diff
@@ -1,2 +1,4 @@
 /build/
-/.vscode/
\ No newline at end of file
+/.vscode/
+state
```
[the remark this answers is not in the document]
Comment (Anttikyll) [reply .gitignore:3]: I'll move this to another PR since it's unrelated.

**Review by TeknoVenus — APPROVED** [review]
LGTM!

**Pull request merged**